lib/chef/provider/package/windows.rb

```diff
@@ -129,6 +129,7 @@ def action_install
         # Chef::Provider::Package action_install + action_remove call install_package + remove_package
         # Pass those calls to the correct sub-provider
         def install_package(name, version)
+          fail Chef::Exceptions::NoWindowsPackageSource, "installer_type '#{installer_type}' requires a source" if new_resource.source.nil?
```

Comment:
> We do in fact require a source for install, the catch is that in the cookbook you could provide a source url as name, and it would work, but wouldn't be idempotent.

In core we do not require source, but w/o this PR the same logic as the cookbook applies defaulting the source to the name.

Not sure I get your comment about the fix being in the cookbook. The goal here is to fix core assuming the cookbook is not used.

I do like using the `requirement.assert` (forgot that existed) over the fail.